src/sas/qtgui/Perspectives/Fitting/FittingPerspective.py

```diff
@@ -349,8 +397,15 @@ def tabCloses(self, index):
         """
         Update local bookkeeping on tab close
         """
+        # update the last-tab closed information
+        # this should be done only for regular fitting
+        if not isinstance(self.tabs[index], ConstraintWidget) and \
```

Comment:
Deleting data, then restoring a tab restores the tab with the deleted data set. This should be handled in `dataDeleted` below.

Reply:
Not really. Deleting data happens from quite a few locations and shouldn't be polluted with the tab recovery code, which is not really deleting the data...